### PR TITLE
Change Error code for undefined in relative comparison - fixes #220

### LINKED ISSUE #220: Change error code for `undefined` in relative comparison
# Problem

https://github.com/dudykr/stc/blob/36f243a4b4b3db1c2838365f695daa5138ac9a07/crates/stc_ts_type_checker/tests/conformance/types/typeRelationships/comparable/equalityStrictNulls.ts#L61-L68

```
Wanted: [RefError { line: 61, column: 13, code: "TS18050" }, RefError { line: 63, column: 13, code: "TS18050" }, RefError { line: 65, column: 14, code: "TS18050" }, RefError { line: 67, column: 14, code: "TS18050" }]
Unwanted: [(61, "TS2532"), (63, "TS2532"), (65, "TS2532"), (67, "TS2532")]', crates/stc_ts_type_checker/tests/tsc.rs:756:17
```

These errors should be replaced with `TS18050` because the **expression** is the identifier reference to the global `undefined`.

# Solution

We may check for `undefined` in https://github.com/dudykr/stc/blob/36f243a4b4b3db1c2838365f695daa5138ac9a07/crates/stc_ts_file_analyzer/src/analyzer/expr/bin.rs#L88 and https://github.com/dudykr/stc/blob/36f243a4b4b3db1c2838365f695daa5138ac9a07/crates/stc_ts_file_analyzer/src/analyzer/expr/bin.rs#L161-L183

After detecting `undefined`s, we **should skip** validation logic at https://github.com/dudykr/stc/blob/36f243a4b4b3db1c2838365f695daa5138ac9a07/crates/stc_ts_file_analyzer/src/analyzer/expr/bin.rs#L617-L620

### PR DESCRIPTION
**Pull request by camc314**
**Description:**

fixes #220 

**BREAKING CHANGE:**

**Related issue (if exists):**

### CODE REVIEW

File: crates/stc_ts_file_analyzer/src/analyzer/expr/bin.rs [hunk 1]
```diff
@@ -613,6 +613,22 @@ impl Analyzer<'_, '_> {
             op!("<=") | op!("<") | op!(">=") | op!(">") => {
                 no_unknown!();
 
+                if let stc_ts_types::Type::Keyword(KeywordType {
```
Comment (littledivy) [inline crates/stc_ts_file_analyzer/src/analyzer/expr/bin.rs:616]: nit
```suggestion
                if let Type::Keyword(KeywordType {
```

File: crates/stc_ts_file_analyzer/src/analyzer/expr/bin.rs [hunk 2]
```diff
@@ -613,6 +613,22 @@ impl Analyzer<'_, '_> {
...
+                    kind: TsKeywordTypeKind::TsUndefinedKeyword,
+                    ..
+                }) = rt
+                {
+                    return Err(Error::UndefinedInRelativeComparison { span: rt.span() });
+                }
+
+                if let stc_ts_types::Type::Keyword(KeywordType {
```
Comment (littledivy) [inline crates/stc_ts_file_analyzer/src/analyzer/expr/bin.rs:624]:
```suggestion
                if let Type::Keyword(KeywordType {
```

File: crates/stc_ts_file_analyzer/src/analyzer/expr/bin.rs (hunk 1 above)
Comment (camc314) [reply crates/stc_ts_file_analyzer/src/analyzer/expr/bin.rs]: oh yeah, thanks :+1:

File: crates/stc_ts_file_analyzer/src/analyzer/expr/bin.rs [hunk 3]
```diff
@@ -613,6 +613,22 @@ impl Analyzer<'_, '_> {
             op!("<=") | op!("<") | op!(">=") | op!(">") => {
                 no_unknown!();
 
+                if let Type::Keyword(KeywordType {
+                    kind: TsKeywordTypeKind::TsUndefinedKeyword,
+                    ..
+                }) = rt
+                {
+                    return Err(Error::UndefinedInRelativeComparison { span: rt.span() });
```
Comment (kdy1) [inline crates/stc_ts_file_analyzer/src/analyzer/expr/bin.rs:621]: I think we should recover from this.

Comment (kdy1) [reply crates/stc_ts_file_analyzer/src/analyzer/expr/bin.rs:621]: Please store the error in `self.storage`

**Review by kdy1 — CHANGES_REQUESTED** [review]

**Review by kdy1 — APPROVED** [review]
Thank you!

**Review by kdy1 — APPROVED** [review]

**Pull request merged**